commercial/app/commercial/CommercialLifecycle.scala

```diff
@@ -0,0 +1,160 @@
+package commercial
+
+import commercial.feeds._
+import common.{AkkaAsync, ExecutionContexts, Jobs, Logging}
+import conf.Configuration.commercial.merchandisingFeedsRoot
+import model.commercial.books.BestsellersAgent
+import model.commercial.jobs.Industries
+import model.commercial.masterclasses.{MasterClassAgent, MasterClassTagsAgent}
+import model.commercial.money.BestBuysAgent
+import model.commercial.travel.{Countries, TravelOffersAgent}
+import model.diagnostics.CloudWatch
+import play.api.{Application => PlayApp, GlobalSettings}
+import services.S3
+
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.util.{Failure, Random, Success}
+
+trait CommercialLifecycle extends GlobalSettings with Logging with ExecutionContexts {
+
+  private val refreshJobs: List[RefreshJob] = List(
+    MasterClassTagsRefresh,
+    MasterclassesRefresh,
+    CountriesRefresh,
+    IndustriesRefresh,
+    MoneyBestBuysRefresh,
+    BooksRefresh,
+    TravelOffersRefresh
+  )
+
+  private def recordEvent(feedName: String, eventName: String, maybeDuration: Option[Duration]): Unit = {
+    val key = s"${feedName.toLowerCase.replaceAll("[\\s/]+", "-")}-$eventName-time"
+    val duration = maybeDuration map (_.toMillis.toDouble) getOrElse -1d
+    CloudWatch.put("Commercial", Map(s"$key" -> duration))
+  }
+
+  override def onStart(app: PlayApp): Unit = {
+
+    val randomFactor = Random.nextInt(15)
+
+    def randomStartSchedule(minsLater: Int = 0) = s"0 ${randomFactor + minsLater}/15 * * * ?"
+
+    def fetchFeed(fetcher: FeedFetcher): Unit = {
+
+      val feedName = fetcher.feedName
+
+      def storeFeed(feed: Feed): Unit =
+        S3.putPrivate(key = s"$merchandisingFeedsRoot/$feedName", value = feed.content, feed.contentType)
```

Comment:
This is persisting our precious object. Should wrap into a function with obvious name that this is special.